001-clarifications-eval.txt

```diff
@@ -0,0 +1,754 @@
+    and removing entries.
+
+    Evaluation of an operation expression starts with an empty stack on
+    which the entries from the initial stack provided by the context are
+    pushed in the order provided. Then the operations are evaluated,
+    starting with the first operation of the stream. Evaluation
+    continues until either an operation has an evaluation error, or
+    until one past the last operation of the stream is reached.
+
+    The result of the evaluation is:
+
+      * If an operation has an evaluation error, or an operation
+        evaluates an expression that has an evaluation error, then the
+        result is an evaluation error.
+
+      * If the current result kind specifies a location description,
+        then:
+
+          * If the stack is empty, the result is a location description
+            with one undefined location description.
+
+            [non-normative] This rule is for backwards compatibility
+            with DWARF Version 5 which uses an empty operation
+            expression for this purpose.
+
+          * If the top stack entry is a location description, or can be
+            converted to one (see 2.5.4.4.3 Memory Location Description
+            Operations), then the result is that, possibly converted,
+            location description. Any other entries on the stack are
+            discarded.
+
+          * Otherwise the DWARF expression is ill-formed.
+
+            > [For further discussion...]
+            > Could define this case as returning an implicit location
+            > description as if the DW_OP_implicit operation is
+            > performed.
+
+      * If the current result kind specifies a value, then:
+
+          * If the top stack entry is a value, or can be converted to
+            one (see 2.5.4.4.3 Memory Location Description Operations),
+            then the result is that, possibly converted, value. Any
+            other entries on the stack are discarded.
+
+          * Otherwise the DWARF expression is ill-formed.
+
+      * If the current result kind is not specified, then:
+
+          * If the stack is empty, the result is a location description
+            with one undefined location description.
+
+            [non-normative] This rule is for backwards compatibility
+            with DWARF Version 5 which uses an empty operation
+            expression for this purpose.
+
+            > [For further discussion...]
+            > This rule is consistent with the rule above for when a
+            > location description is requested. However, GDB appears to
+            > report this as an error and no GDB tests appear to cause
+            > an empty stack for this case.
+
+          * Otherwise, the top stack entry is returned. Any other
+            entries on the stack are discarded.
+
+    An operation expression is encoded as a byte block with some form of
+    prefix that specifies the byte count. It can be used:
```

Comment:
When we specify a list in normative text, I think we imply that the enumeration is exhaustive, i.e. there are no other usages of DWARF expressions allowed.  I would not define such a list as it is bound to get stale when new usages are added.

Reply:
I believe there is value in providing the list and not letting it get stale. There are numerous other places in the standard that list things (and some of those lists are in fact stale and later parts fix them).

As a reader of the standard it was helpful to me to know how the major concepts are used and have cross references to where they are defined.